Source/Core/Core/HW/WiimoteEmu/DesiredWiimoteState.cpp

```diff
@@ -115,33 +107,16 @@ SerializedWiimoteState SerializeDesiredState(const DesiredWiimoteState& state)
     std::visit(
         [&s](const auto& arg) {
           using T = std::decay_t<decltype(arg)>;
-          if constexpr (!std::is_same_v<std::monostate, T>)
-          {
-            static_assert(sizeof(arg) <= 6);
-            static_assert(std::is_trivially_copyable_v<T>);
-            std::memcpy(&s.data[s.length], &arg, sizeof(arg));
-            s.length += sizeof(arg);
-          }
+          static_assert(sizeof(arg) <= 6);
+          Common::BitCastPtr<T>(&s.data[s.length]) = arg;
```

Comment:
This is not valid if `sizeof(arg) == 0`, which I think happens in the no extension case?

Reply:
Actually wait, this case probably never happens due to the `if (extension)` check up there, which means the previous `if constexpr (!std::is_same_v<std::monostate, T>)` was redundant. So that's fine then.

Reply:
In the no extension case, `arg` will be `std::monostate` which has a `sizeof` 1 but the code will not be reached because of the `if (extension)` just above on line 105.